src/include/libplacebo/vulkan.h

```diff
@@ -265,11 +265,13 @@ VkImage pl_vulkan_unwrap(const struct pl_gpu *gpu, const struct pl_tex *tex,
 // access mode specified by the user, and fire the given semaphore (required!)
 // when this is done. This marks the image as held. Attempting to perform any
 // pl_tex_* operation (except pl_tex_destroy) on a held image is an error.
+// If `export` is true, the image will additionally be transitioned to the
+// external queue family.
 //
 // Returns whether successful.
 bool pl_vulkan_hold(const struct pl_gpu *gpu, const struct pl_tex *tex,
                     VkImageLayout layout, VkAccessFlags access,
-                    VkSemaphore sem_out);
+                    bool export, VkSemaphore sem_out);
```

Comment:
As discussed on IRC, I don't think it actually makes that much sense to have the export as part of the `pl_vulkan_hold`. Rather we should have a `pl_vulkan_export` (analog to `pl_buf_export`). Rationale:

1. The VkImageLayout is always going to be `VK_IMAGE_LAYOUT_GENERAL` anyway
2. The `VkAccessFlags` are meaningless for a queue family transition

Plus, that removes the dependency on vulkan-specific information, so we could in theory move this to `pl_gpu` in an abstract way (similar to `pl_buf_export`) by introducing some sort of `pl_gpu_semaphore` wrapper for externally shared semaphores.

Reply:
That being said, it probably still makes sense to *treat* `pl_vulkan_export` like a hold-type operation, i.e. `pl_vulkan_release` will release it - since unlike the buffer case we can't do this implicitly due to the fact that we synchronize all texture access using semaphores.

Reply:
Maybe `pl_vulkan_hold_external`?

Reply:
I've implemented `pl_vulkan_hold_external` and `pl_vulkan_release_external` to encapsulate the layout and access mode handling. I want to leave the semaphore handling as-is and deal with that in a separate change to introduce exported semaphores. I'll try and introduce a semaphore wrapper as part of that.